src/script/lua_api/l_client.cpp

```diff
@@ -414,6 +443,59 @@ int ModApiClient::l_get_csm_restrictions(lua_State *L)
 	return 1;
 }
 
+// store_texture(texture, path)
+int ModApiClient::l_store_texture(lua_State *L)
+{
+	const std::string texturename = luaL_checkstring(L, 1);
+	const std::string filename = luaL_checkstring(L, 2);
+	
+	if (filename.empty())
+		return 0;
+
+	lua_rawgeti(L, LUA_REGISTRYINDEX, CUSTOM_RIDX_CURRENT_MOD_NAME);
+	std::string current_mod = readParam<std::string>(L, -1, "");
+	if (current_mod.empty()) {
+		current_mod = getScriptApiBase(L)->getOrigin();
+		if (current_mod.empty())
+			return 0;
+	}
+
+	std::string base_path = fs::AbsolutePath(getClient(L)->getClientModsLuaPath()) + DIR_DELIM + current_mod;
```

Comment:
This might be the wrong path if you do not run minetest in place, For instance, on my computer this would be in `/usr/local/share/minetest/clientmods/`

Reply:
What does `fs::AbsolutePath(getClient(L)->getClientModsLuaPath())` return for you? I'm thinking it should work.

Reply:
it returns `/usr/local/share/minetest/clientmods/online_craftguide`, while the mod is in `~/.minetest/clientmods/`

Reply:
If I interpret it correctly, `getClient(L)->getClientModsLuaPath()` is the problem, since it only returns the main dir. 

Here's my system info:
```
gpcf@knoblauch ~/.minetest $ minetest --version
Minetest 5.3.0-dev-5dd073a91-dirty (Linux)
Using Irrlicht 1.8.4
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_GETTEXT=1
USE_SOUND=1
USE_CURL=1
USE_FREETYPE=1
USE_LUAJIT=1
STATIC_SHAREDIR="/usr/local/share/minetest"
STATIC_LOCALEDIR="/usr/local/share/locale"

```

Reply:
You apparently have to place your mods in your sharedir, as `ClientModConfiguration modconf(getClientModsLuaPath());` is how CSMs are loaded, and that is `porting::path_share + DIR_DELIM + "clientmods";`

Reply:
Should be addressed by the latest commit, try placing your mods in your user directory.